### PR TITLE
set up netlify for book previews

### PR DESCRIPTION
**Pull request by JessicaS11**

### CODE REVIEW

**Comment by JessicaS11** [comment]
There are two ways to set up Netlify. One is to use Netlify's build services (free version has 300 minutes/month build time) via the GitHub app. This results in the bot comments (I deleted them) and checks shown in this PR. The other, which relies instead on the higher GitHub actions build minutes limit, is to do the build on GitHub via an action (which is what this PR adds). As per [this helpful article,](https://marekpukaj.medium.com/build-with-github-actions-host-on-netlify-ebf5fa505616) I added the required secrets to get the action to run. Since it's a new action, we'll have to merge it before we can try running it!

File: .github/workflows/netlifypreview.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: NetlifyPreview
+
+on:
+  pull_request_target:
+    types: [labeled, synchronize]
+
+jobs:
+  add-preview:
+    runs-on: ubuntu-20.04
```
Comment (weiji14) [inline .github/workflows/netlifypreview.yml:9]: Run on this year's Ubuntu.
```suggestion
    runs-on: ubuntu-22.04
```

File: .github/workflows/netlifypreview.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        deploy-message: "Deploy from GitHub Actions"
+        enable-commit-comment: false
+        enable-pull-request-comment: true
+        overwrites-pull-request-comment: true
+        alias: deploy-preview-${{ github.event.number }}
+      env:
+        NETLIFY_AUTH_TOKEN: ${{ secrets.NETLIFY_AUTH_TOKEN }}
+        NETLIFY_SITE_ID: ${{ secrets.NETLIFY_SITE_ID }}
+      timeout-minutes: 1
```
Comment (weiji14) [inline .github/workflows/netlifypreview.yml:37]: Would a 1 minute timeout be a bit too short? How about 3 min?

```suggestion
      timeout-minutes: 3

```

**Review by weiji14 — APPROVED** [review]
> There are two ways to set up Netlify. One is to use Netlify's build services (free version has 300 minutes/month build time) via the GitHub app. This results in the bot comments (I deleted them) and checks shown in this PR. The other, which relies instead on the higher GitHub actions build minutes limit, is to do the build on GitHub via an action (which is what this PR adds). As per [this helpful article,](https://marekpukaj.medium.com/build-with-github-actions-host-on-netlify-ebf5fa505616) I added the required secrets to get the action to run. Since it's a new action, we'll have to merge it before we can try running it!

Thanks Jessica! Agree with going with GitHub Actions instead of the GitHub App (which always felt buggy to me). Feel free to merge after the suggested changes, I see that this uses `pull_request_target`, so yes, need to merge before we know if it works!

**Comment by weiji14** [comment]
Have made a small patch at #8 to install `jupyter-book` in the `.github/workflows/netlifypreview.yml` workflow so that the documentation build works.

**Pull request merged**